src/pages/companion/profile-temp/[externalUserId]/index.tsx

```diff
@@ -0,0 +1,199 @@
+import { use, useEffect, useState } from "react";
+import Profile from '../../../../components/companion/blueprintProfiles/profile' // update these once done testing
+import ExtraInfo from "../../../../components/companion/blueprintProfiles/extraInfo";
+import { useRouter } from "next/router";
+import CompanyInfo from "../../../../components/companion/blueprintProfiles/companyInfo";
+import AttendeeInfo from "../../../../components/companion/blueprintProfiles/attendeeInfo";
+import { TopNav } from '../../../../components/companion/navigation/top-nav';
+import { SideNav } from '../../../../components/companion/navigation/side-nav';
+import { BottomNav } from '../../../../components/companion/navigation/bottom-nav';
+import { motion } from 'framer-motion';
+
+
+
+interface UserProfile {
+    name: string;
+    role: string;
+    hobby: string;
+    linkedIn: string;
+    funFacts: string[];
+    interests: string[];
+    additionalLinks: string[];
+    profilePicUrl: string;
+    companyLogoUrl?: string;
+    company?: string;
+    major?: string;
+    year?: string;
+}
+
+// THEN GO THROUGH AND TEST IT WITH MUTLIPLE INTERESTS AND DIFFERENT LENGTHS FOR EACH TO CHECK BOUNDARIES
+// FIX RESPONSIVENESS FOR S8+, 
+
+// NOTES FOR PR:
+// - S8+ mobile screens needed to reformat the desing because it was too squished
+// - right now, the linkedin page links are expecting with the https://www included, but we can also do some simple regex to 
+//   refactor them
+// - did not include the bullet points on the fun facts and interests because it made the spacing weird when the word lengths became longer
+// - figma design font was smaller, but I think it may not be readable on all screens so I increased the size. It may look a little more squished as a result
+// - edited animated border to have the children match the sizing of the border - can revert if it breaks anything else
+
+/*
+ THIS IS THE COMPONENT FOR A STUDENT USER PROFILE
+
+ MY PLAN FOR THE UI IS TO HAVE OUR INDEX.TSX PAGE RENDER BASED ON THE USER RETURNED BY THE API CALL
+
+ EX FLOW. 
+ - USER TAPS NFC CARD WHICH LOADS THIS LINK WITH ENDPOINT OF A USERID ** still need to implement the grabbing from url
+ - MAKE API CALL FOR INFO ON USER 
+ - IF STUDENT PROFILE, RENDER THIS COMPONENT, ELSE RENDER DELEGATE PROFILE
+
+ *Note* - on a delegate profile, if the 'visit-page' button is clicked for the company, route to company profile 
+ - turn both company and user into this component and conditionally render based on role
+ - could make a company page route? or make this one conditionally render even more - i think i like own route better though
+*/
+
+const UserProfilePage = () => {
+    const [userData, setUserData] = useState<UserProfile | null>(null);
+    const [isLoading, setIsLoading] = useState(true);
+    const [isSideNavOpen, setIsSideNavOpen] = useState(false);
+    const router = useRouter();
+    const [user, setUser] = useState<string>('ubcbiztech');  // Default user fallback
```

Comment:
Any reason why 'ubcbiztech' is our fallback? What should the expected value be when we get this running in prod?

Reply:
hmm, yea I guess when I first started thinking about it I thought if no user was given, a fallback to our club profile page could be an option, but I think that's probably unnecessary. When we actually run this the expected value should be an external user Id and will be used to fetch the profile data. I believe my component will be rendered from isaac's nfc card logic. My understanding is that when the nfc card is scanned (or someone types in a link manually), it will search for the user and render either the requested profile or the scanner's profile depending on whether or not the requested profile exists. In this case I think we should always have a user value.